File: api/app/api/routes/summaries.py
from uuid import UUID

from fastapi import APIRouter, Depends, Path
from sqlalchemy.orm import Session

from app.db.session import get_db
from app.schemas.summaries import SummaryResponse, SummaryViewedResponse
from app.services.summaries import get_or_create_summary, mark_summary_viewed

router = APIRouter()


@router.get("/sessions/{sessionId}/summary", response_model=SummaryResponse)
def read_session_summary(
    session_id: UUID = Path(alias="sessionId"),
    db: Session = Depends(get_db),
) -> SummaryResponse:
    return get_or_create_summary(db, session_id)


@router.post(
    "/sessions/{sessionId}/summary/viewed",
    response_model=SummaryViewedResponse,
)
def mark_session_summary_viewed(
    session_id: UUID = Path(alias="sessionId"),
    db: Session = Depends(get_db),
) -> SummaryViewedResponse:
    return mark_summary_viewed(db, session_id)

File: web/src/api/summaries.ts
import { requestJson } from "./client";
import type { SummaryResponse, SummaryViewedResponse } from "../types/summary";

export function getSummary(sessionId: string): Promise<SummaryResponse> {
  return requestJson<SummaryResponse>(`/sessions/${encodeURIComponent(sessionId)}/summary`);
}

export function markSummaryViewed(sessionId: string): Promise<SummaryViewedResponse> {
  return requestJson<SummaryViewedResponse>(
    `/sessions/${encodeURIComponent(sessionId)}/summary/viewed`,
    {
      method: "POST"
    }
  );
}
